UefiTestingPkg/AuditTests/DMAProtectionAudit/UEFI/VTd/DmarAcpiTable.c

```diff
@@ -198,7 +198,11 @@ GetDmarAcpiTableRmrr (
     switch (DmarHeader->Type) {
       case EFI_ACPI_DMAR_TYPE_RMRR:
         // If RMRR found add to end of linked list
-        NewNode       = (RMRRListNode *)AllocateZeroPool (sizeof (RMRRListNode));
+        NewNode = (RMRRListNode *)AllocateZeroPool (sizeof (RMRRListNode));
+        if (NewNode == NULL) {
+          return NULL;
+        }

```

Comment:
Similar to the function in IVRSAcpiTable.c, it would be preferable to clean up the list on the failure return path.